src/webui/src/static/model/searchspace.ts

```diff
@@ -0,0 +1,185 @@
+import { SingleAxis, MultipleAxes, TableObj } from '../interface';
+
+
+const SUPPORTED_SEARCH_SPACE_TYPE = [
+    'choice',
+    'layer_choice',
+    'input_choice',
+    'randint',
+    'uniform',
+    'quniform',
+    'loguniform',
+    'qloguniform',
+    'normal',
+    'qnormal',
+    'lognormal',
+    'qlognormal'
+];
+
+
+function fullNameJoin(prefix: string, name: string): string {
+    return prefix ? (prefix + '/' + name) : name;
+}
+
+class NumericAxis implements SingleAxis {
+    min: number = 0;
+    max: number = 0;
+    type: string;
+    name: string;
+    fullName: string;
+    scale: 'log' | 'linear';
+    nested = false;
+
+    constructor(name: string, fullName: string, type: string, value: any) {
+        this.name = name;
+        this.fullName = fullName;
+        this.type = type;
+        this.scale = type.indexOf('log') !== -1 ? 'log' : 'linear';
+        if (type === 'randint') {
+            this.min = value[0];
+            this.max = value[1] - 1;
+        } else if (type.indexOf('uniform') !== -1) {
+            this.min = value[0];
+            this.max = value[1];
+        } else if (type.indexOf('normal') !== -1) {
+            const [mu, sigma] = [value[0], value[1]];
+            this.min = mu - 4 * sigma;
```

Comment:
why x 4

Reply:
In normal distribution, [mu - 4*sigma, mu + 4*sigma] is a very safe domain of the values.